src/qibocal/cli/report.py

```diff
@@ -18,8 +20,16 @@
 
 def generate_figures_and_report(
     node: Completed, target: Union[QubitId, QubitPairId, list[QubitId]]
-):
-    """Returns figures and table for report."""
+) -> tuple[str, list[go.Figure]]:
+    """Calling protocol plot by checking if fit has been performed.
+
+    Args:
+        node (Completed): completed node
+        target (Union[QubitId, QubitPairId, list[QubitId]]): protocol target
+
+    Returns:
+        tuple[str, list[go.Figure]]: report outcome
+    """
```

Comment:
This is a nice example, so I'm just saying for this reason.

I would make the docs way more succinct, e.g. by saving all the parameters' types, since already present in the signature.
Moreover, they could also be made much more human-readable, by avoiding the fixed template
```suggestion
    """Calling protocol plot by checking if fit has been performed.
    
    It operates on a completed `node` and a specific protocol `target`, generating
    a report outcome (cf. `ReportOutcome`).
    """
```

And describe what is a `ReportOutcome` in the docstring of such an alias (just define `ReportOutcome = tuple[str, list[go.Figure]]` above).

This is just a proposal, not necessarily a request for this PR. But the idea is to reduce the SNR even in our files :P
(so, avoid the parts that could be autogenerated, and let Sphinx do that work)

Reply:
I agree with the fact that we should not duplicate documentation.
In my case I just installed an extension that will generate the docstring for me that's why I put it there :)
Since it is already there I prefer to keep it, once we unify the notation among all libraries we can update all docstrings.

Reply:
> In my case I just installed an extension that will generate the docstring for me that's why I put it there :)

I had something similar. Then, I thought: "but if it's able to generate it, why should I save it to Git at all?", and started writing it more concise.

> Since it is already there I prefer to keep it, once we unify the notation among all libraries we can update all docstrings.

It was just a proposal, to start moving in that direction.

Reply:
> It was just a proposal, to start moving in that direction.

Actually having a `ReportOutcome` makes sense. I've also updated the docstrings :)